syncopy/preproc/resampling.py

```diff
@@ -0,0 +1,137 @@
+
+    Parameters
+    ----------
+    data  : (N, K) :class:`numpy.ndarray`
+        Uniformly sampled multi-channel time-series data
+        The 1st dimension is interpreted as the time axis,
+        columns represent individual channels.
+    orig_fs : float
+        The original sampling rate
+    new_fs : float
+        The target sampling rate after resampling
+    lpfreq : None or float, optional
+        Leave at `None` for standard anti-alias filtering with
+        the new Nyquist or set explicitly in Hz
+        If set to `-1` use SciPy's default kaiser windowed FIR
+    order : None or int, optional
+        Order (length) of the firws anti-aliasing filter.
+        The default `None` will create a filter of
+        maximal order which is the number of samples in the trial.
+
+    Returns
+    -------
+    resampled : (N, K) :class:`~numpy.ndarray`
+        The resampled signals
+
+    See also
+    --------
+    `SciPy's resample implementation <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.resample_poly.html>`_
+    syncopy.preproc.compRoutines.downsample_cF : Straightforward and cheap downsampling
+    """
+
+    nSamples = data.shape[0]
+
+    # get up/down sampling factors
+    up, down = _get_updn(orig_fs, new_fs)
+    fs_ratio = new_fs / orig_fs
+
+    # -- design firws low-pass filter --
+
+    # default cuts at new Nyquist
+    if lpfreq is None:
+        f_c = 0.5 * fs_ratio
+    # for backend tests only,
+    # negative values don't pass the frontend                
+    elif lpfreq == -1:
+        f_c = None
+    # explicit cut-off
+    else:
+        f_c = lpfreq / orig_fs
+    if order is None:
+        order = nSamples
+
+    if f_c:
+        # filter has to be applied to the upsampled data    
+        window = firws.design_wsinc("hamming",
+                                  order=nSamples,
+                                  f_c=f_c / up)
+    else:
+        window = ('kaiser', 5.0)   # SciPy default
+    
+    resampled = sci_sig.resample_poly(data, up, down, window=window, axis=0)
+
+    return resampled
+
+
+def downsample(
+    dat,
+    samplerate=1,
+    new_samplerate=1,
+):
+    """
+    Provides basic downsampling of signals. The `new_samplerate` should be
+    an integer division of the original `samplerate`.
+
+    Parameters
+    ----------
+    dat : (N, K) :class:`numpy.ndarray`
+        Uniformly sampled multi-channel time-series data
+        The 1st dimension is interpreted as the time axis,
+        columns represent individual channels.
+    samplerate : float
+        Sample rate of the input data
+    new_samplerate : float
+        Sample rate of the output data
+
+    Returns
+    -------
+    resampled : (X, K) :class:`~numpy.ndarray`
+        The downsampled data
+
+    """
+
+    # we need integers for slicing
+    skipped = int(samplerate // new_samplerate)
+
+    return dat[::skipped]
+
+
+def _get_updn(orig_fs, new_fs):
```

Comment:
Using the `fractions` module for computing the sampling factors is really clever!

Reply:
I think I got it from SO 😃 But yeah, `np.gcd` did not do the trick directly..